### PR TITLE
feat: add metrics for use of transform override for CAD

### PR DESCRIPTION
**Pull request by larsmoa**
# Description

- Track usage of CAD node transform.
- Enable batch requests in Mixpanel to avoid flooding with stats by accident

# Checklist:

Here is a checklist that should completed before merging this given feature. 
Any shortcomings from the items below should be explained and detailed within the contents of this PR.

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.
Deemed not necessary - tested on three different browsers on two OS, one desktop and one mobile.

### CODE REVIEW

File: viewer/packages/metrics/src/MetricsLogger.ts [hunk 1]
```diff
@@ -115,6 +120,56 @@ export class MetricsLogger {
...
+    (nodeCount: number, matrix: THREE.Matrix4) => MetricsLogger.trackCadNodeTransformOverriddenImpl(nodeCount, matrix),
+    MetricsLogger.TrackCadNodeTransformOverriddenThrottleDelay
+  );
+
+  private static trackCadNodeTransformOverriddenImpl(nodeCount: number, matrix: THREE.Matrix4): void {
+    const { zeroVector, oneVector, identityRotation, translation, scale, rotation } =
+      MetricsLogger.trackCadNodeTransformOverriddenVars;
+    matrix.decompose(translation, rotation, scale);
+
+    const hasRotation = Math.abs(rotation.dot(identityRotation) - 1.0) > 1e-5; // Angle between two quaternions is arccos(q1.dot(q2))
```
Comment (haakonflatval-cognite) [inline viewer/packages/metrics/src/MetricsLogger.ts:149]: I don't think this formula is right. I believe we want what's described here: https://se.mathworks.com/matlabcentral/answers/415936-angle-between-2-quaternions . In essence, I believe the formula should be

$$\theta = 2 cos^{-1}((q_1 \times q_2^{-1}).w)$$
Where the cross denotes quaternion multiplication

**Review by haakonflatval-cognite — CHANGES_REQUESTED** [review]
Had one issue, otherwise looks good

File: viewer/packages/metrics/src/MetricsLogger.ts (hunk 1 above)
Comment (larsmoa) [reply viewer/packages/metrics/src/MetricsLogger.ts:149]: Fixed.

Comment (haakonflatval-cognite) [reply viewer/packages/metrics/src/MetricsLogger.ts:149]: Okay, I guess it's good enough :P

**Review by haakonflatval-cognite — APPROVED** [review]
Fine :+1:

**Pull request merged**